os/pm/pm_initialize.c

```diff
@@ -74,6 +82,119 @@
 
 struct pm_global_s g_pmglobals;
 
+/* Sleep state count threshold array. These values are used to enter the next reduced power state */
+int g_pm_switch_thresholds[CONFIG_PM_NSTATE];
+
+/* Sleep state activity threshold array. These values are used to enter any sleep state */
+int g_pm_enter_thresholds[CONFIG_PM_NSTATE];
+
+/* Sleep state activity threshold array. These values are used to exit any sleep state */
+int g_pm_exit_thresholds[CONFIG_PM_NSTATE];
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: pm_parse_thresholds
+ *
+ * Description:
+ *   Parse the switch, enter & exit threshold values to the global arrays.
+ *   If not provided by the SoC initilaize them with default values.
+ *
+ ****************************************************************************/
+
+static void pm_parse_thresholds(int array_type, const char *thr_str, int threshold_array[])
+{
+	int index = 0;
+	bool assign_defaultvalue = true;
+
+	/* Initialize state threshold value to be read from thresh_str */
+	int threshold = 0;
+	/* Variable to track number of tick count threshold provided by the SOC */
+	int check_count = 0;
+
+	/* Parse the sleep state tick count thresholds from the Kconfig */
+
+	while (thr_str && *thr_str) {
+
+		/* Extract thresholds from string input in config as integer value */
+		threshold = strtoul(thr_str, NULL, 0);
+
+		while (*thr_str != ',' && *thr_str) {
+			thr_str++;
+		}
+
+		if (*thr_str == ',') {
+			thr_str++;
+		}
+
+		check_count++;
+
+		/* Case when more state threshold values are provided by PM_STATE_COUNT_THRESH*/
+		if (check_count > (CONFIG_PM_NSTATE - 1)) {
+			DEBUGASSERT(!(check_count > (CONFIG_PM_NSTATE - 1)));
+		}
+
+		/* Update sleep state threshold array with extracted integer value */
+		threshold_array[index++] = threshold;
+	}
+
+	/* Check if all the state entering tick count thresholds are provided by the SOC */
+
+	if (check_count == (CONFIG_PM_NSTATE - 1)) {
+
+		/* Case when all state threshold values are provided by PM_STATE_COUNT_THRESH*/
+		assign_defaultvalue = false;
```

Comment:
The assign_defaultvalue variable is not required. 
Please move the complete code in the "if" section at line 161 directly to this location.